pkg/operator/prunecontroller/prune_controller.go

```diff
@@ -0,0 +1,232 @@
+package prune
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1informer "k8s.io/client-go/informers/core/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// PruneController is a controller that watches the operand pods and deletes old
+// revisioned secrets that are not used anymore.
+type PruneController struct {
+	targetNamespace string
+	secretPrefixes  []string
+
+	secretGetter   corev1client.SecretsGetter
+	podGetter      corev1client.PodsGetter
+	podInformer    corev1informer.PodInformer
+	secretInformer corev1informer.SecretInformer
+
+	cachesToSync  []cache.InformerSynced
+	queue         workqueue.RateLimitingInterface
+	eventRecorder events.Recorder
+}
+
+const (
+	pruneControllerWorkQueueKey = "key"
+	numOldRevisionsToPreserve   = 5
+)
+
+// NewPruneController creates a new pruning controller
+func NewPruneController(
```

Comment:
why do we need a new controller?

Reply:
because we have none? The one from static pods is super complex and does much much more. I looked into generalizing it, but decided it is not worth it.

Reply:
@sttts will this controller move to library-go ?